src/Symfony/Component/Mime/Security/SMimeSigner.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mime\Security;
+
+use Symfony\Component\Mime\Exception\RuntimeException;
+use Symfony\Component\Mime\Message;
+use Symfony\Component\Mime\WrappedMessage;
+
+/**
+ * @author Sebastiaan Stok <s.stok@rollerscapes.net>
+ *
+ * @experimental in 4.3
+ */
+final class SMimeSigner extends SMime
+{
+    private $signCertificate;
+    private $signPrivateKey;
+    private $signOptions;
+    private $extraCerts;
+
+    /**
+     * @see https://secure.php.net/manual/en/openssl.pkcs7.flags.php
+     *
+     * @param string       $certificate
+     * @param string|array $privateKey  If the key needs an passphrase use array('file-location', 'passphrase') instead
+     * @param int          $signOptions Bitwise operator options for openssl_pkcs7_sign()
+     * @param string       $extraCerts  A file containing intermediate certificates needed by the signing certificate
+     */
+    public function __construct(string $certificate, $privateKey, int $signOptions = PKCS7_DETACHED, $extraCerts = null)
+    {
+        $this->signCertificate = $this->normalizeFilePath($certificate);
+
+        if (\is_array($privateKey)) {
+            $this->signPrivateKey = $privateKey;
+            $this->signPrivateKey[0] = $this->normalizeFilePath($privateKey[0]);
```

Comment:
Would it be an idea to add some validation on the array structure, or make an object to show it?

Reply:
Or an extra argument itself, but I'm not sure which is more preferred. 
Not using a password is something I'm highly against as this file needs to be secured somehow, _you can provide the file as-is and then provide the password as a secret._

Reply:
I settled for an extra argument, it's more likely this is used than not as a private-key should be stored safely. And it makes the arguments more related.